Makefile

```diff
@@ -21,6 +21,19 @@ for file in /etc/os-release /usr/lib/os-release; do \
     fi \
 done)
 
+GOARCH=$(shell go env GOARCH)
+HOST_ARCH=$(shell arch)
+
+ifeq ($(ARCH),)
+	ARCH = $(GOARCH)
+endif
```

Comment:
just fmi, do these changes implement cross compilation, or do we always expect a native compile?
For cross-compilation, I was sort of expecting to see something like:
```
else
 if ARCH != GOARCH
   GOARCH = ARCH
```

to pass the requested `ARCH` (if different from the host) on to the golang build. Maybe I'm missing something though?

Reply:
These changes don't take any account of cross-compiling. The only reason I added `arch` output is to show the "real" host architecture along with the standard golang name for that same arch.

If we want to do cross-compiling, that's a bigger topic and should be handled on a different PR imho.